src/tools/haxelib/Main.hx

```diff
@@ -1341,42 +1341,51 @@ class Main {
 		print("Library "+prj+" current version is now "+version);
 	}
 
-    function checkRec( prj : String, version : String, l : List<{ rep : String, project : String, version : String, info : Infos }> ) {
-        var reps = getRepositories();
-        var found = false;
-        for (rep in reps) {
-		    var pdir = rep + Data.safe(prj);
-                if( !FileSystem.exists(pdir) ) {
-                    continue;
-                }
-                found = true;
-		var version = if( version != null ) version else getCurrent(pdir);
-		var vdir = pdir + "/" + Data.safe(version);
```

Comment:
This is the fixed logic.  `getCurrent(pdir)` throws an exception if `.current` file does not exit.